client/collection.go

```diff
@@ -0,0 +1,160 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/tigrisdata/tigris-client-go/driver"
+	"github.com/tigrisdata/tigris-client-go/filter"
+	"github.com/tigrisdata/tigris-client-go/projection"
+	"github.com/tigrisdata/tigris-client-go/schema"
+	"github.com/tigrisdata/tigris-client-go/update"
+)
+
+var (
+	errNotFound = fmt.Errorf("document not found")
```

Comment:
let's start thinking about these errors as well. Better to do it early otherwise we need to define helpers like this.